### PR TITLE
Don't expose JSON in libosrm but C++ object

### PR DESCRIPTION
**Pull request by TheMarex**
# Issue

Explores https://github.com/Project-OSRM/osrm-backend/issues/2029. Goal is to get some benchmarking numbers to get a compelling argument here.

## Tasklist
- [ ] Port a hacked node-osrm
- [ ] Run benchmarks

### CODE REVIEW

**Comment by ssuluh** [comment]
+1, this will be great since my project using libosrm directly.

File: include/engine/guidance/result.hpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+{
+namespace guidance
+{
+class Waypoint;
+class Route;
+
+struct Result
+{
+    std::vector<Waypoint> waypoints;
+    std::vector<Route> routes;
```
Comment (daniel-j-h) [inline include/engine/guidance/result.hpp]: Just had a look over this changeset - what you're doing here is not allowed: you're storing incomplete types in stdlib containers. The container have to know the `value_type`'s size though.

File: include/engine/guidance/route.hpp [hunk 2]
```diff
@@ -1,17 +1,30 @@
...
 namespace guidance
 {
 
+struct RouteLeg;
+
 struct Route
 {
     double duration;
     double distance;
+    std::vector<RouteLeg> legs;
```
Comment (daniel-j-h) [inline include/engine/guidance/route.hpp]: Same here

**Comment by TheMarex** [comment]
Outdated.

**Pull request closed without merging**